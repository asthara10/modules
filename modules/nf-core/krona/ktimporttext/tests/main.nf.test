nextflow_process {

    name "Test Process KRONA_KTIMPORTTEXT"
    script "../main.nf"
    process "KRONA_KTIMPORTTEXT"

    tag "modules"
    tag "modules_nfcore"
    tag "krona"
    tag "krona/ktimporttext"

    test("test_krona_ktimporttext_multi") {


        when {
            process {
                """
                
                input[0] = [
        [ id:'test', single_end:false ], // meta map
        [
            file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/delete_me/krona/ktimporttext.txt', checkIfExists: true), // krona default test file
            file(params.test_data['sarscov2']['metagenome']['kraken_report'], checkIfExists: true), //Kraken2 report file
            file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/delete_me/krona/kaiju_out4krona.txt', checkIfExists: true) // Kaiju output 4 krona
        ]
    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("test_krona_ktimporttext_single") {
        when {
            process {
                """
                        input[0] = [
        [ id:'test', single_end:false ], // meta map
        [
            file('http://krona.sourceforge.net/examples/text.txt', checkIfExists: true) // krona default test file
        ]
    ]


                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
                { assert snapshot(process.out).match() }
            )
        }
    }
    
}
