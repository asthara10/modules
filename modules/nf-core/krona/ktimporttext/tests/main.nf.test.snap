{
    "versions": {
        "content": [
            [
                "versions.yml:md5,b8847bf9b5b00d34a38cf6f774c4c5db"
            ]
        ],
        "timestamp": "2023-11-24T13:26:18.217477"
    }
}